modules/synthetics/main.tf

```diff
@@ -4,16 +4,39 @@ locals {
   enabled            = module.this.enabled
   alert_tags         = local.enabled && var.alert_tags != null ? format("%s%s", var.alert_tags_separator, join(var.alert_tags_separator, var.alert_tags)) : ""
   datadog_synthetics = { for k, v in var.datadog_synthetics : k => v if local.enabled }
+
+  all_public_locations = [
+    "aws:ap-northeast-1",
+    "aws:ap-northeast-2",
+    "aws:ap-south-1",
+    "aws:ap-southeast-1",
+    "aws:ap-southeast-2",
+    "aws:ca-central-1",
+    "aws:eu-central-1",
+    "aws:eu-north-1",
+    "aws:eu-west-1",
+    "aws:eu-west-2",
+    "aws:eu-west-3",
+    "aws:sa-east-1",
+    "aws:us-east-2",
+    "aws:us-west-1",
+    "aws:us-west-2",
+    "azure:eastus"
+  ]
+
+  public_locations = contains(split(",", lower(join(",", var.public_locations))), "all") ? local.all_public_locations : var.public_locations
```

Comment:
If no public locations were provided, couldnt we just default to all ?

Reply:
why check for contains? If we provide `all`, it should be just one item "all" in the list?

Reply:
Agreed

Reply:
unresolved this because the public locations still do a split join when andriys solution is simpler